File: services/ts-express-superstruct/services/check_service.ts
import * as service from './../check_service'
import * as models from './../models'

export let checkService = (): service.CheckService => {
    let checkEmpty = async (): Promise<void> => {
        throw new Error('Not Implemented')
    }

    let checkQuery = async (params: service.CheckQueryParams): Promise<void> => {
        throw new Error('Not Implemented')
    }

    let checkUrlParams = async (params: service.CheckUrlParamsParams): Promise<void> => {
        throw new Error('Not Implemented')
    }

    let checkForbidden = async (): Promise<service.CheckForbiddenResponse> => {
        throw new Error('Not Implemented')
    }

    let sameOperationName = async (): Promise<service.SameOperationNameResponse> => {
        throw new Error('Not Implemented')
    }

    return {checkEmpty, checkQuery, checkUrlParams, checkForbidden, sameOperationName}
}

File: services/scala-play/services/services/check/CheckService.scala
package services.check

import javax.inject._
import scala.concurrent._
import models._

@Singleton
class CheckService @Inject()()(implicit ec: ExecutionContext) extends ICheckService {
  override def checkEmpty(): Future[CheckEmptyResponse] = Future { ??? }
  override def checkForbidden(): Future[CheckForbiddenResponse] = Future { ??? }
  override def sameOperationName(): Future[SameOperationNameResponse] = Future { ??? }
}
